File: frontend/codex-llm-ui/src/requests/get_stream.ts
async function get_stream(prompt: string) {
    const url = "http://localhost:8000/stream/"

    const response = await fetch(url, {
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/x-ndjson'
    },
    body: JSON.stringify({ "prompt": prompt }),
    })

    return response
}

export default get_stream

File: backend/app/modules/get_stream.py
import httpx
from configs.api_config import TIMEOUT
from typing import AsyncGenerator
import httpx

async def get_stream(
    data: dict,
    headers: dict,
    url: str
) -> AsyncGenerator[str, None]:
    async with httpx.AsyncClient() as client:
        async with client.stream(
            "POST",
            url=url,
            timeout=httpx.Timeout(TIMEOUT),
            headers=headers,
            json=data,
        ) as response:
            response.raise_for_status()
            async for chunk in response.aiter_text():
                yield chunk